ARCs/arc-0023.md

```diff
@@ -0,0 +1,76 @@
+---
+
+## Abstract
+
+The following document introduces a convention for appending information to the compiled application's bytes. 
+The goal of this convention is to standardize the process of verifying and interacting with smart contracts. The encoded byte is `arc23` followed by the length of the CID representing a folder of files & the CID itself.
+
+## Specification
+
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in <a href="https://www.ietf.org/rfc/rfc2119.txt">RFC-2119</a>.
+
+> Comments like this are non-normative.
+
+### CID
+The <a href="https://github.com/multiformats/cid">CID</a> allows to access the corresponding files using <a href="https://docs.ipfs.tech/">IPFS</a>.
+
+The CID **MUST** represent one file or a folder of files.
+The minimum requirement for this ARC is a file representing the ABI (ABI as described in [ARC-4](arc-0004.md))
+
+### Encoding
+The encoded byte is `arc23` followed by the length of the CID, the number of files, and the CID itself.
+
+eg: for the example stored in `/asset/arc-0023/Folder`, 
+`arc23 59 2 bafybeihdzhfifv46p27ee6weqq6h7ydttzwni7gvdogr3mg6rwscdyzici`
+`bafybeihdzhfifv46p27ee6weqq6h7ydttzwni7gvdogr3mg6rwscdyzici` correspond to this Folder: `../assets/arc-0023/Folder`. It can be obtained with `ipfs add --cid-version=1 Folder/*  -q -w -n| tail -1`
+
+If there is only one file, the CID **MUST** be related to the ABI of the application. 
+eg: `arc23 59 1 bafkreiajguyi3i5hf4h7blslsv53rjbexzzvpkyhdf26hmuqthrk4iko4a`
+`bafkreiajguyi3i5hf4h7blslsv53rjbexzzvpkyhdf26hmuqthrk4iko4a` correspond to this Abi File: `../assets/arc-0023/Folder/Abi.json`. It can be obtained with `ipfs add --cid-version=1 Folder/Abi.json -q`
+
+> The option -n is for obtaining the hash without pinning the file, you **MAY** remove the option in production
+
+Otherwise, if there is more than one file, the CID must correspond to a folder of files. The ABI **MUST** be accessible by querying and IPFS node with `{cid}/Abi.json`
+eg:`bafybeihdzhfifv46p27ee6weqq6h7ydttzwni7gvdogr3mg6rwscdyzici/Abi.json`.
+
+### Appending
+The encoded byte is appended to the compiled application as a <a href="https://developer.algorand.org/docs/get-details/dapps/avm/teal/opcodes/#bytecblock-bytes">bytecblock</a> containing 1-byte constant which is the encoded object.
+> The reason to use `bytecblock` is that adding a `bytecblock` opcode at the end of a TEAL program does not change the semantics of the program, as long as opcodes are correctly aligned, there is no jump after the last position (that would make the program fail without `bytecblock`), and there is enough space left to add the opcode.
+The size of the compiled application + the bytecblock should be, at most, the maximum size of a compiled application according to the latest consensus parameters supported by the compiler.
+
```

Comment:
MAybe add a paragraph about non-deployed application.

### Use in development

In development, it may be preferable to not publish the application information on IPFS.
Tools that need these information (such as local block explorer) can request this information as a .zip or .tar.gz containing at the root all the required files.